src/Compilers/Core/Portable/SourceGeneration/GlobalAliases.cs

```diff
@@ -31,6 +31,22 @@ public static GlobalAliases Create(ImmutableArray<(string aliasName, string symb
         return aliasAndSymbolNames.IsEmpty ? Empty : new GlobalAliases(aliasAndSymbolNames);
     }
 
+    public static GlobalAliases Create(ImmutableArray<GlobalAliases> aliasesArray)
+    {
+        if (aliasesArray.Length == 0)
+            return Empty;
+
+        if (aliasesArray.Length == 1)
+            return aliasesArray[0];
```

Comment:
avoid intermediary allocations for the common case of just a single file with global aliases.